src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -962,6 +962,28 @@ public void setNeedsTool(final IToolType neededTool, final int minimalLevel)
         this.needsToolLevel = minimalLevel;
     }
 
+    @Override
+    public boolean equals(final Object o)
+    {
+        if(!(o instanceof AbstractBuilding))
+        {
+            return false;
+        }
+
+        if(((AbstractBuilding) o).getID().equals(this.getID()))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1126)